### PR TITLE
CLI: taskrun command

### LINKED ISSUE #242: CLI: taskrun command
**Description**

Implement `Run()` method of the `kcp taskrun` command which runs the given command (which can also be a script, or other program) for specific SKRs.

Steps performed:

1. Resolve the target specifiers to runtimes using `GardenerRuntimeResolver` and a custom InstanceLister based on the list runtimes API exposed in kyma/control-plane#180 (also leveraged by skr runtime list command). 
2. Using `ParallelOrchestrationStrategy` with the amount of workers specified as –-parallelism, execute the following for each runtime:
    - Check if SKR kubeconfig is present in kubeconfig-cache-dir. If not, obtain the kubeconfig from the kubeconfig-service-url
    - Run the specified COMMAND [ARGS ...] having the SKR kubeconfig exported in the KUBECONFIG variable.
    - Capture lines written to standard output and error, prefix lines with Shoot name.
    - Save exit status
3. Wait for all workers to finish, and exit with an aggregated exit status: non-zero in case of command failure , 0 if all command runs finished successfully.

**Acceptance Criteria**
- Refactor [orchestration.InstanceLister](https://github.com/kyma-project/control-plane/blob/master/components/kyma-environment-broker/internal/orchestration/resolver.go#L29-L31) interface to `RuntimesLister`, which returns list of [`runtime.RuntimeDTO`](https://github.com/kyma-project/control-plane/blob/master/components/kyma-environment-broker/internal/runtime/model.go#L9). Implement orchestration.RuntimeLister which operates the same way as the [`getRuntimes` handler](https://github.com/kyma-project/control-plane/blob/master/components/kyma-environment-broker/internal/runtime/handler.go#L48)
- Implement runtime lister in CLI which is based on the runtime.Client getting the runtimes info remotely from KEB 
- Implement steps to perform the input command for resolved SKRs.

### PR DESCRIPTION
**Pull request by ebensom**
**Description**

CLI: taskrun command

Changes proposed in this pull request:

- Implement taskrun command based on `orchestrarion` `GardenerRuntimeResolver` and `ParallelOrchestrationtrategy`
- Refactor CLI logger to be based on logrus

**Related issue(s)**

Resolves #242

### CODE REVIEW

**Review by lumi017 — DISMISSED** [review]

File: docs/cli/commands/kcp_taskrun.md [hunk 1]
```diff
@@ -33,9 +33,11 @@ kcp taskrun --target {TARGET SPEC} ... [--target-exclude {TARGET SPEC} ...] COMM
 ## Options
 
 ```
-      --keep                         Option that allows you to keep downloaded kubeconfig files after execution for caching purposes.
+      --keep-kubeconfig              Option that allows you to keep downloaded kubeconfig files after execution for caching purposes.
       --kubeconfig-dir string        Directory to download Runtime kubeconfig files to. By default, it is a random-generated directory in the OS-specific default temporary directory (e.g. /tmp in Linux).
-  -p, --parallelism int              Number of parallel commands to execute. (default 8)
+      --no-kubeconfig                Option to turn off the downloading and exposure of kubeconfig for each runtime.
+      --no-prefix-output             Option to omit prefixing each output line with the Runtime name. By default, all output lines are prepended for better traceability.
```
Comment (klaudiagrz) [inline docs/cli/commands/kcp_taskrun.md:39]:
```suggestion
      --no-prefix-output             Option that omits the prefixing of each output line with the Runtime name. By default, all output lines are prepended for better traceability.
```

File: docs/cli/commands/kcp_taskrun.md [hunk 2]
```diff
@@ -33,9 +33,11 @@ kcp taskrun --target {TARGET SPEC} ... [--target-exclude {TARGET SPEC} ...] COMM
...
+  -p, --parallelism int              Number of parallel commands to execute. (default 4)
```
Comment (klaudiagrz) [inline docs/cli/commands/kcp_taskrun.md:40]:
```suggestion
  -p, --parallelism int              Number of parallel commands to execute. The default value is 4.
```

File: docs/cli/commands/kcp_taskrun.md [hunk 3]
```diff
@@ -53,13 +55,14 @@ kcp taskrun --target {TARGET SPEC} ... [--target-exclude {TARGET SPEC} ...] COMM
 ```
       --config string                Path to the KCP CLI config file. Can also be set using the KCPCONFIG environment variable. Defaults to $HOME/.kcp/config.yaml .
       --gardener-kubeconfig string   Path to the kubeconfig file of the corresponding Gardener project which has permissions to list/get Shoots. Can also be set using the KCP_GARDENER_KUBECONFIG environment variable.
+      --gardener-namespace string    Gardener namespace (project) to use. Can also be set using the KCP_GARDENER_NAMESPACE environment variable.
```
Comment (klaudiagrz) [inline docs/cli/commands/kcp_taskrun.md:58]:
```suggestion
      --gardener-namespace string    Gardener Namespace (project) to use. Can also be set using the KCP_GARDENER_NAMESPACE environment variable.
```

File: docs/cli/commands/kcp_upgrade.md [hunk 4]
```diff
@@ -10,13 +10,14 @@ Performs upgrade operations on Kyma Runtimes.
 ```
       --config string                Path to the KCP CLI config file. Can also be set using the KCPCONFIG environment variable. Defaults to $HOME/.kcp/config.yaml .
       --gardener-kubeconfig string   Path to the kubeconfig file of the corresponding Gardener project which has permissions to list/get Shoots. Can also be set using the KCP_GARDENER_KUBECONFIG environment variable.
+      --gardener-namespace string    Gardener namespace (project) to use. Can also be set using the KCP_GARDENER_NAMESPACE environment variable.
```
Comment (klaudiagrz) [inline docs/cli/commands/kcp_upgrade.md:13]:
```suggestion
      --gardener-namespace string    Gardener Namespace (project) to use. Can also be set using the KCP_GARDENER_NAMESPACE environment variable.
```

File: docs/cli/commands/kcp_upgrade_kyma.md [hunk 5]
```diff
@@ -45,13 +45,14 @@ kcp upgrade kyma --target {TARGET SPEC} ... [--target-exclude {TARGET SPEC} ...]
 ```
       --config string                Path to the KCP CLI config file. Can also be set using the KCPCONFIG environment variable. Defaults to $HOME/.kcp/config.yaml .
       --gardener-kubeconfig string   Path to the kubeconfig file of the corresponding Gardener project which has permissions to list/get Shoots. Can also be set using the KCP_GARDENER_KUBECONFIG environment variable.
+      --gardener-namespace string    Gardener namespace (project) to use. Can also be set using the KCP_GARDENER_NAMESPACE environment variable.
```
Comment (klaudiagrz) [inline docs/cli/commands/kcp_upgrade_kyma.md:48]:
```suggestion
      --gardener-namespace string    Gardener Namespace (project) to use. Can also be set using the KCP_GARDENER_NAMESPACE environment variable.
```

File: docs/cli/commands/kcp_taskrun.md (hunk 2 above)
Comment (ebensom) [reply docs/cli/commands/kcp_taskrun.md:40]: Default value in parenthesis is auto-generated.

File: tools/cli/pkg/command/taskrun.go [hunk 6]
```diff
@@ -44,29 +91,272 @@ For each subprocess, the following Runtime-specific data are passed as environme
...
+			return errors.Wrap(err, "while creating temporary kubeconfig directory")
+		}
+		cmd.kubeconfingDirTemp = true
+	}
+
+	// Validate task command
+	if _, err := exec.LookPath(args[0]); err != nil {
+		return err
+	}
+	cmd.taskCommand = exec.CommandContext(cmd.cobraCmd.Context(), args[0], args[1:]...)
```
Comment (fhivemind) [inline tools/cli/pkg/command/taskrun.go:173]: Unneccessarry expansion
```go
cmd.taskCommand = exec.CommandContext(cmd.cobraCmd.Context(), args...)
```

**Review by klaudiagrz — APPROVED** [review]

File: tools/cli/pkg/logger/logger.go [hunk 7]
```diff
@@ -1,35 +1,35 @@
...
-	"log"
-	"os"
-
 	"github.com/int128/kubelogin/pkg/adaptors/logger"
+	"github.com/sirupsen/logrus"
 	"github.com/spf13/pflag"
 )
 
+// CfgLevel is the configured logging level
+var CfgLevel int
```
Comment (fhivemind) [inline tools/cli/pkg/logger/logger.go:10]: Is this being configured anywhere?

File: tools/cli/pkg/command/taskrun.go (hunk 6 above)
Comment (ebensom) [reply tools/cli/pkg/command/taskrun.go:173]: It's not unnecessary. The called function's signature is `func exec.CommandContext(ctx context.Context, name string, arg ...string) *exec.Cmd`

So caller must assign the second argument, and may pass subsequent arguments.

**Review by lumi017 — APPROVED** [review]

**Pull request merged**